core/state_transition.go

```diff
@@ -741,6 +763,19 @@ func (st *stateTransition) refundGas(refundQuotient uint64) uint64 {
 		st.evm.Config.Tracer.OnGasChange(st.gasRemaining, 0, tracing.GasChangeTxLeftOverReturned)
 	}
 
+	if optimismConfig := st.evm.ChainConfig().Optimism; optimismConfig != nil && !st.msg.IsDepositTx {
+		// Return ETH to transaction sender for operator cost overcharge.
+		if actualCost := st.evm.Context.OperatorCostFunc(new(big.Int).SetUint64(st.gasUsed()), st.evm.Context.Time); actualCost != nil {
+			if worstCaseCost != nil {
+				wcU256, worstCaseCostOverflow := uint256.FromBig(worstCaseCost)
+				acU256, actualCostOverflow := uint256.FromBig(actualCost)
+				if !actualCostOverflow && !worstCaseCostOverflow {
```

Comment:
This logic where an overflow happens and no balance change happens is part of the spec. we would need to make sure that reth exactly follows this. i dont buy that not doing anything is the best way to handle this

Reply:
https://github.com/ethereum-optimism/op-geth/pull/388/files#r1941998578

Reply:
if we cap here: https://github.com/ethereum-optimism/op-geth/pull/388/files#r1942000197 then i think we dont need to worry about the overflow cases. we could panic on overflow if we are super sure that it cannot be triggered

Reply:
We have established that the highest operator fee is an int with 77 bits, so an overflow is not possible. Does that cover this discussion?